File: frontend/src/app/model/workflow-operator.ts
import { AbstractBaseItem } from './base-item';

export interface WorkflowOperator extends AbstractBaseItem {
  /**
   * UUID of enclosed item
   */
  itemId: string;

  /**
   * name of the operator
   */
  name: string;

  /**
   * y position
   */
  posY: number;

  /**
   * x position
   */
  posX: number;
}

File: backend/src/main/java/org/hetida/designer/backend/model/WorkflowOperator.java
package org.hetida.designer.backend.model;


import lombok.Data;
import org.hetida.designer.backend.enums.ItemType;

import javax.persistence.*;
import java.util.UUID;

@Entity
@Data
@Table(name = "workflowoperator")
public class WorkflowOperator {

    @Id
    @Column(name = "id", columnDefinition = "uuid")
    private UUID id;

    @Column(name = "itemid", columnDefinition = "uuid")
    private UUID ItemId;

    @Column(name = "name")
    private String name;

    @Column(name = "type")
    @Enumerated(EnumType.STRING)
    private ItemType type;

    @Column(name = "posy")
    private int posY;

    @Column(name = "posx")
    private int posX;
}




